README.md

```diff
@@ -1,2 +1,150 @@
+
+Mailto [did document]s are also verifiable offline, when [domain key][] of the email address is known, implying significantly less network requests than with most other DIDs.
+
+## The `did:mailto` Format
+
+The format for the `did:key` method conforms to the [DID-CORE] specification and is encoding of the [email address]. It consists of the `did:mailto:` prefix, followed by the `domain` part of an email address, `:` character and percent encoded `local-part` of the email address.
+
+The ABNF definition can be found below. The formal rules describing valid `domain-name` syntax are described in [RFC1035], [RFC1123], [RFC2181]. The `domain-name` and `user-name` corresponds to `domain` and `local-part` respectively of the email address described in [RFC2822]. All "mailto" DIDs MUST conform to the DID Syntax ABNF Rules.
+
+```abnf
+did       = "did:mailto:" domain-name ":" user-name
+user-name = 1*idchar
+idchar    = ALPHA / DIGIT / "." / "-" / "_" / pct_enc
+pct_enc   = "%" HEXDIG HEXDIG
+```
+
+##### EXAMPLE 1. A jsmith@example.com
+
+```
+did:mailto:example.com:jsmith
+```
+
+##### EXAMPLE 2 tag+alice@web.mail
+
+```
+did:mailto:web.mail:tag%2Balice
+```
+
+## Operations
+
+The following section outlines the DID operations for the `did:mailto` method.
+
+### Create (Register)
+
+There is intentionally no specification for publishing `did:mailto` documents as single source of truth are not implied. Same `did:mailto` identifier MAY (intentionally) correspond to different [DID document] in different sessions.
+
+Creating a `did:mailto` document is done by:
+
+1. Creating a [did:key] identifier.
+2. Sending an email from [DID Subject] email address with a [key authentication] message in the subject.
+
+#### Key Authentication
+
+Key authentication message must conform to the following ABNF definition, which refers to `did-key-format` defined in [did:key] specification
+
+```abnf
+auth = "I am also known as " did-key-format
+```
+
+##### EXAMPLE 3
+
+```
+I am also known as did:key:z6MkhaXgBZDvotDkL5257faiztiGiC2QtKLGpbnnEGta2doK
+```
+
+### Read (Resolve)
+
+Specification intentionally does not specify how to resolve [DKIM-Signature] for the given `did:mailto` identifier as those could be arranged out of bound between participants. Optional programmatic resolution over [SMTP] could be defined in the future iterations.
+
+The following steps MUST be executed to resolve the [DID document] from the [DKIM-Signature]:
+
+1. Extract [key authentication] from the `subject` header.
+1. Extract [did:key] from the extracted [key authentication].
+1. Extract sender email address from the `from` header.
+1. Resolve [DID document] from extracted [did:key].
+1. Set `id` of the [DID document] to the `did:mailto` identifier of the sender email address.
+1. Set `alsoKnownAs` of the document to extracted [did:key].
+
+### Deactivate (Revoke)
```

Comment:
> One of the biggest reasons to revoke would be because the user lost access to the email address. Is there a way to revoke in this case?

Note that revocation here implies revoking a key as opposed to revoking a whatever permissions email has, which is I think what you’re thinking of. If I don’t have access to the email, I may want to revoke it’s privileges but that is to be done at another layer - delegation to the email address. That is out of scope here.

Reason you’d want to revoke here is because you’ve lost a private key, or perhaps key got compromised. 

> How do I prevent any other user from claiming my did:key:x? Is it correct that we don't have to...because anything I send with issuer as did:mailto:x has to be signed with the private key for did:key:x.

If someone say evil@attacker.com associates my key with their email they are essentially granting me all the privileges that email address has, which is opposite of what they would want so there is no initiative to do this.

> Assuming that is true, anyone could claim they are known as did:key:x, but they wouldn't be able to act as did:key:x because they don't have the private key.

I think you could put it that way.

> Without signed proof, you can't revoke the use of did:key:x (since we don't know you have the private key) but it could revoke the use of the senders email address as an alias of did:key:x.

I think you’re thinking of association in opposite direction.

If I send you an email saying X is my key it’s reasonable to expect that messages signed by that key is from the owner of that email. If I then send you a message (from the same email address) saying I no longer use that X key it is reasonable to expect that if you see message signed with that key X it’s probably an attacker trying to impersonate me.

Here you trust that messages are from trusted source (email address) and what you verify is the DKIM signature to ensure that email is indeed from the owner of the address.

Reply:
Ok I understand thanks for the clarification.